lib/src/http_body_impl.dart

```diff
@@ -19,118 +19,96 @@ class HttpBodyHandlerTransformer
 
   @override
   Stream<HttpRequestBody> bind(Stream<HttpRequest> stream) {
-    return Stream<HttpRequestBody>.eventTransformed(
-        stream,
-        (EventSink<HttpRequestBody> sink) =>
-            _HttpBodyHandlerTransformerSink(_defaultEncoding, sink));
-  }
-}
-
-class _HttpBodyHandlerTransformerSink implements EventSink<HttpRequest> {
-  final Encoding _defaultEncoding;
-  final EventSink<HttpRequestBody> _outSink;
-  int _pending = 0;
-  bool _closed = false;
-
-  _HttpBodyHandlerTransformerSink(this._defaultEncoding, this._outSink);
-
-  @override
-  void add(HttpRequest request) {
-    _pending++;
-    HttpBodyHandlerImpl.processRequest(request, _defaultEncoding)
-        .then(_outSink.add, onError: _outSink.addError)
-        .whenComplete(() {
-      _pending--;
-      if (_closed && _pending == 0) _outSink.close();
-    });
-  }
-
-  @override
-  void addError(Object error, [StackTrace stackTrace]) {
-    _outSink.addError(error, stackTrace);
-  }
-
-  @override
-  void close() {
-    _closed = true;
-    if (_pending == 0) _outSink.close();
+    var pending = 0;
+    var closed = false;
```

Comment:
where does this get set `true`? should it now be in `handleDone`?

Reply:
yes, should be set in `handleDone`. Apparently there are no tests...

Reply:
Added a regression test and fixed the bug. PTAL.